pylint_django/__init__.py

```diff
@@ -3,10 +3,17 @@
 
 import sys
 
-from pylint_django import plugin
+from pylint_django import plugin, compat
 
 if sys.version_info < (3, ):
     raise DeprecationWarning("Version 0.11.1 was the last to support Python 2. "
                              "Please migrate to Python 3!")
 
-register = plugin.register  # pylint: disable=invalid-name
+# Older versions of pylint does not support load_configuration() hook. To be
+# backward compatible we call load_configuration() directly from register()
+# hook. Using legacy approach yields issue #181.
+# For more details see issue #181 and PR #214.
+if compat.REGISTER_LEGACY:
```

Comment:
-1 on this syntax for the following reasons:

* `__init__.py` should be as simple as possible
* `db_performance.py::register()` calls `load_configuration()` based on this condition. Why should `plugin.py::register()` be different ? 
* Also consider maintainability and other contributors down the road: While `register()` is a function provided by pylint's API interface `register_legacy()` is not. Let's save ourselves the troubles of figuring out what that was or even missing it entirely sometime down the road. (these things happen, I can show you a very recent example where I screwed up a PR and then an entire team of maintainers failed to see my mistake, pushed new version upstream and screwed up even more. We also had a similar less spectacular scenario in pylint_django recently.).

To be clear leave `__init__.py` alone and put the if condition inside `plugin.register()` the same way you do in `db_performance.register()`.